client/network/src/service.rs

```diff
@@ -844,6 +840,34 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 		}
 	}
 
+	/// Variation of `request` which can connect to peers when necessary.
+	///
+	/// In some cases we do request data, without prior notification. For those cases this function
+	/// exists. It has no guarantees on connection staying alive, so you might still want to pass
+	/// false to the `connect` parameter and ensure liveness of connections via peer sets.
+	/// Connecting to a peer can take up to several seconds, if a peer exposes multiple
+	/// addresses, but is only reachable by some of them, for example.
+	///
+	/// Apart from exposing whether or not we want to connect, in case of a disconnected peer, this
+	/// function also takes a sender for sending the response back, instead of returning the
+	/// response directly.
+	pub fn send_request(
```

Comment:
```suggestion
	pub fn request_channel(
```

I really don't like `send_request` because it doesn't mean much :grimacing: and in particular doesn't show any difference in what it does compared to `request`.

Reply:
I took inspiration [here](https://github.com/paritytech/substrate/blob/b2b9af4f1553fd82c98a816486e9ed0e117104e5/client/network/src/request_responses.rs#L274) for the name, as it has the exact same interface :-) 

Not sure about `request_channel` though, as it sounds like requesting a channel or alternatively like a request channel, when in fact we have a channel for responses. I will try to come up with something.